rclpy/src/rclpy/duration.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__DURATION_HPP_
+#define RCLPY__DURATION_HPP_
+
+#include <pybind11/pybind11.h>
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+/// Create a duration
+/**
+ * On failure, an exception is raised and NULL is returned if:
+ *
+ * Raises RuntimeError on initialization failure
+ * Raises TypeError if argument of invalid type
+ * Raises OverflowError if nanoseconds argument cannot be converted to int64_t
```

Comment:
I don't think it raises OverflowError anymore. It also may raise MemoryError (via std::bad_alloc()) instead of RuntimeError.

Reply:
Ah, good catch. That's why I had to update `duration.py`...hadn't thought to update the docs.